cborg/src/Codec/CBOR/Pretty.hs

```diff
@@ -161,45 +163,63 @@ pprint = do
   hexRep term
   str " "
   case term of
-    TkInt      i  TkEnd -> ppTkInt i
-    TkInteger  i  TkEnd -> ppTkInteger i
-    TkWord     w  TkEnd -> ppTkWord w
-    TkBytes    bs TkEnd -> ppTkBytes bs
-    TkBytesBegin  TkEnd -> ppTkBytesBegin
-    TkString   t  TkEnd -> ppTkString t
-    TkStringBegin TkEnd -> ppTkStringBegin
-    TkListLen  w  TkEnd -> ppTkListLen w
-    TkListBegin   TkEnd -> ppTkListBegin
-    TkMapLen   w  TkEnd -> ppTkMapLen w
-    TkMapBegin    TkEnd -> ppTkMapBegin
-    TkBreak       TkEnd -> ppTkBreak
-    TkTag      w  TkEnd -> ppTkTag w
-    TkBool     b  TkEnd -> ppTkBool b
-    TkNull        TkEnd -> ppTkNull
-    TkSimple   w  TkEnd -> ppTkSimple w
-    TkFloat16  f  TkEnd -> ppTkFloat16 f
-    TkFloat32  f  TkEnd -> ppTkFloat32 f
-    TkFloat64  f  TkEnd -> ppTkFloat64 f
-    TkEncoded  _  TkEnd -> ppTkEncoded
-    TkEnd               -> str "# End of input"
+    TkInt      i   TkEnd     -> ppTkInt i
+    TkInt64    i   TkEnd     -> ppTkInt64 i
+    TkInteger  i   TkEnd     -> ppTkInteger i
+    TkWord64   w   TkEnd     -> ppTkWord64 w
+    TkWord     w   TkEnd     -> ppTkWord w
+    TkBytes    bs  TkEnd     -> ppTkBytes bs
+    TkBytesBegin   TkEnd     -> ppTkBytesBegin
+    TkByteArray ba TkEnd     -> ppTkByteArray ba
+    TkUtf8ByteArray ba TkEnd -> ppTkUtf8ByteArray ba
+    TkString   t   TkEnd     -> ppTkString t
+    TkStringBegin  TkEnd     -> ppTkStringBegin
+    TkListLen  w   TkEnd     -> ppTkListLen w
+    TkListBegin    TkEnd     -> ppTkListBegin
+    TkMapLen   w   TkEnd     -> ppTkMapLen w
+    TkMapBegin     TkEnd     -> ppTkMapBegin
+    TkBreak        TkEnd     -> ppTkBreak
+    TkTag      w   TkEnd     -> ppTkTag w
+    TkTag64    w   TkEnd     -> ppTkTag64 w
+    TkBool     b   TkEnd     -> ppTkBool b
+    TkNull         TkEnd     -> ppTkNull
+    TkUndef        TkEnd     -> ppTkUndef
+    TkSimple   w   TkEnd     -> ppTkSimple w
+    TkFloat16  f   TkEnd     -> ppTkFloat16 f
+    TkFloat32  f   TkEnd     -> ppTkFloat32 f
+    TkFloat64  f   TkEnd     -> ppTkFloat64 f
+    TkEncoded  _   TkEnd     -> ppTkEncoded
+    TkEnd                    -> str "# End of input"
     _ -> fail $ unwords ["pprint: Unexpected token:", show term]
 
 ppTkInt        :: Int        -> PP ()
 ppTkInt i = str "# int" >> parens (shown i)
 
+ppTkInt64      :: Int64      -> PP ()
+ppTkInt64 i = str "# int64" >> parens (shown i)
+
 ppTkInteger    :: Integer    -> PP ()
 ppTkInteger i = str "# integer" >> parens (shown i)
 
+ppTkWord64     :: Word64     -> PP ()
+ppTkWord64 w = str "# word64" >> parens (shown w)
+
 ppTkWord       :: Word       -> PP ()
 ppTkWord w = str "# word" >> parens (shown w)
 
+ppTkByteArray  :: SlicedByteArray -> PP ()
+ppTkByteArray bs = str "# bytearray" >> parens (shown $ sizeofSlicedByteArray bs)
+
+ppTkUtf8ByteArray  :: SlicedByteArray -> PP ()
+ppTkUtf8ByteArray bs = str "# utf8bytearray" >> parens (shown $ sizeofSlicedByteArray bs)
```

Comment:
There's no formal definition of this format, but if we follow the cbor.me example, then I think we would call the byte array just "bytes" and the utf8 byte array just "text". Since these are not different CBOR types, just different Haskell representations for CBOR bytes or text.